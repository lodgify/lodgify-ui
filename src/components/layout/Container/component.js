import React from 'react';
import { Container as SemanticContainer } from 'semantic-ui-react';

/**
 * Container is the Lodgify UI interface for the
 * Semantic UI Container.
 * @returns {Object}
 */
export const Component = ({ textAlign, ...props }) => (
  <SemanticContainer {...props} textAlign={textAlign} />
);

Component.displayName = 'Container';
