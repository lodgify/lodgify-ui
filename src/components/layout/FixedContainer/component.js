import React from 'react';
import PropTypes from 'prop-types';

/**
 * A container for placing children in a fixed position relative to the viewport.
 */
// eslint-disable-next-line jsdoc/require-jsdoc
export const Component = ({ bottom, children, left, right, top, zIndex }) => (
  <div
    style={{
      bottom,
      display: 'inline-block',
      left,
      position: 'fixed',
      right,
      top,
      zIndex,
    }}
  >
    {children}
  </div>
);

Component.displayName = 'FixedContainer';

Component.defaultProps = {
  bottom: '',
  left: '',
  right: '',
  top: '',
  zIndex: 1,
};

Component.propTypes = {
  /** The distance between the container's bottom edge and the bottom edge of the viewport. */
  bottom: PropTypes.string,
  /** The children to be wrapped with a fixed container. */
  children: PropTypes.node.isRequired,
  /** The distance between the container's left edge and the left edge of the viewport. */
  left: PropTypes.string,
  /** The distance between the container's right edge and the right edge of the viewport. */
  right: PropTypes.string,
  /** The distance between the container's top edge and the top edge of the viewport. */
  top: PropTypes.string,
  /** The priority of the fixed container in the stacking context of the page. */
  zIndex: PropTypes.number,
};
