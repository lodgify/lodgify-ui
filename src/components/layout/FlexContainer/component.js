import React from 'react';
import PropTypes from 'prop-types';

/**
 * A container for laying out children using the
 * [flexbox method](https://developer.mozilla.org/en-US/docs/Learn/CSS/CSS_layout/Flexbox).
 * Fills the full width and height of its parent or remaining space of a flex parent.
 */
// eslint-disable-next-line jsdoc/require-jsdoc
export const Component = ({
  alignContent,
  alignItems,
  children,
  flexDirection,
  flexWrap,
  justifyContent,
}) => (
  <div
    className="flex-container"
    style={{
      display: 'flex',
      flexGrow: '1',
      height: '100%',
      alignContent,
      alignItems,
      flexDirection,
      flexWrap,
      justifyContent,
    }}
  >
    {children}
  </div>
);

Component.displayName = 'FlexContainer';

Component.defaultProps = {
  alignContent: null,
  alignItems: null,
  flexDirection: null,
  flexWrap: null,
  justifyContent: null,
};

Component.propTypes = {
  /** If there is more than one line of children, the alignment of the children as a whole relative to the cross axis of the flex container. */
  alignContent: PropTypes.oneOf([
    'flex-start',
    'flex-end',
    'center',
    'space-between',
    'space-around',
    'stretch',
  ]),
  /** The alignment of each child relative to the cross axis of the line on which that child is rendered. */
  alignItems: PropTypes.oneOf([
    'flex-start',
    'flex-end',
    'center',
    'baseline',
    'stretch',
  ]),
  /** The children to be wrapped with a flex container. */
  children: PropTypes.node.isRequired,
  /** The orientation of the main axis of the flex container. */
  flexDirection: PropTypes.oneOf([
    'row',
    'row-reverse',
    'column',
    'column-reverse',
  ]),
  /** The wrap behaviour of the children. */
  flexWrap: PropTypes.oneOf(['nowrap', 'wrap', 'wrap-reverse']),
  /** The alignment of the children as a whole relative to the main axis of the flex container. */
  justifyContent: PropTypes.oneOf([
    'flex-start',
    'flex-end',
    'center',
    'space-between',
    'space-around',
    'space-evenly',
  ]),
};
