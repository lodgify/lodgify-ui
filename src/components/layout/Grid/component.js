import React from 'react';
import PropTypes from 'prop-types';
import { Grid } from 'semantic-ui-react';

/**
 * Grid is the Lodgify UI interface for the
 * Semantic UI Grid.
 *
 * To customise how the Grid gets computed, override the @columnCount
 * definition of Semantic UI.
 */
// eslint-disable-next-line jsdoc/require-jsdoc
export const Component = ({ areColumnsCentered, isStackable, ...props }) => (
  <Grid {...props} centered={areColumnsCentered} stackable={isStackable} />
);

Component.displayName = 'Grid';

Component.defaultProps = {
  areColumnsCentered: false,
  isStackable: false,
};

Component.propTypes = {
  /** Are the columns in the grid centered. */
  areColumnsCentered: PropTypes.bool,
  /** Do the items in the grid stack on top of one another on mobile devices. */
  isStackable: PropTypes.bool,
};
