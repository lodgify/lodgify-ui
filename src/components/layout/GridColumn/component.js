import React from 'react';
import PropTypes from 'prop-types';
import { Grid } from 'semantic-ui-react';

/**
 * GridColumn is the Lodgify UI interface for the
 * Semantic UI Grid.Column.
 *
 * @returns {Object}
 */
export const Component = ({ verticalAlignContent, ...props }) => (
  <Grid.Column {...props} verticalAlign={verticalAlignContent} />
);

Component.displayName = 'GridColumn';

Component.defaultProps = {
  verticalAlignContent: 'top',
  width: 12,
};

Component.propTypes = {
  /** Vertically align the content of the column to the bottom, middle or top. */
  verticalAlignContent: PropTypes.oneOf(['bottom', 'middle', 'top']),
  /** The width of the column */
  width: PropTypes.oneOf([1, 2, 3, 4, 5, 6, 7, 8, 9, 10, 11, 12]),
};
