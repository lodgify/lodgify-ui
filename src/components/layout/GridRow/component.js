import React from 'react';
import PropTypes from 'prop-types';
import { Grid } from 'semantic-ui-react';

/**
 * GridRow is the Lodgify UI interface for the
 * Semantic UI Grid.Row.
 */
// eslint-disable-next-line jsdoc/require-jsdoc
export const Component = ({ horizontalAlignContent, ...props }) => (
  <Grid.Row {...props} textAlign={horizontalAlignContent} />
);

Component.displayName = 'GridRow';

Component.defaultProps = {
  horizontalAlignContent: 'left',
};

Component.propTypes = {
  /** Horizontally align the content of the row to the left, center or right. Text content can be justified to fit the width of the row. */
  horizontalAlignContent: PropTypes.oneOf([
    'left',
    'center',
    'right',
    'justified',
  ]),
};
