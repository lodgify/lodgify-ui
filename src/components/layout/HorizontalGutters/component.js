import React from 'react';
import { Container } from 'semantic-ui-react';

/**
 * HorizontalGutters wraps child components with responsive
 * horizontal gutters on the left and right.
 */
// eslint-disable-next-line jsdoc/require-jsdoc
export const Component = props => <Container {...props} />;

Component.displayName = 'HorizontalGutters';
