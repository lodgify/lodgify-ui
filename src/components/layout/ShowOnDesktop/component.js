import React from 'react';
import PropTypes from 'prop-types';
import getClassNames from 'classnames';

/**
 * ShowOnDesktop wraps child components and hides
 * them on all other devices
 */
// eslint-disable-next-line jsdoc/require-jsdoc
export const Component = ({ children, parent, parentProps }) =>
  React.createElement(
    parent,
    {
      ...parentProps,
      className: getClassNames(parentProps.className, 'show-on-desktop'),
    },
    children
  );

Component.defaultProps = {
  children: null,
  parent: 'div',
  parentProps: {},
};

Component.displayName = 'ShowOnDesktop';

Component.propTypes = {
  /** The child components only to be rendered on desktop  */
  children: PropTypes.node,
  /** The parent component that will be rendered. */
  parent: PropTypes.oneOfType([PropTypes.node, PropTypes.func]),
  /** The props to be passed to the parent component. */
  // eslint-disable-next-line react/forbid-prop-types
  parentProps: PropTypes.object,
};
