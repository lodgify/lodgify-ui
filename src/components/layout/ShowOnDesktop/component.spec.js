import React from 'react';
import { shallow } from 'enzyme';
import { Responsive } from 'semantic-ui-react';
import {
  expectComponentToHaveProps,
  expectComponentToBe,
  expectComponentToHaveDisplayName,
} from '@lodgify/enzyme-jest-expect-helpers';

import { Component as ShowOnDesktop } from './component';

const props = {
  parent: 'section',
  parentProps: { width: 'ðŸš¸' },
  children: 'div',
};

const getShowOnDesktop = () => shallow(<ShowOnDesktop {...props} />);

describe('<ShowOnDesktop />', () => {
  it('should render a single Semantic UI `Responsive` component', () => {
    const wrapper = getShowOnDesktop();

    expectComponentToBe(wrapper, Responsive);
  });

  describe('the `Responsive` component', () => {
    it('it should get the right props', () => {
      const wrapper = getShowOnDesktop();

      expectComponentToHaveProps(wrapper, {
        as: props.parent,
        children: props.children,
      });
    });
  });

  it('should have `displayName` `ShowOnDesktop`', () => {
    expectComponentToHaveDisplayName(ShowOnDesktop, 'ShowOnDesktop');
  });
});
