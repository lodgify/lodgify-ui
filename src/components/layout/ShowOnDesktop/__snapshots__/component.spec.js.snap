// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ShowOnDesktop /> it should render the right structure 1`] = `
<ShowOnDesktop
  parent="section"
  parentProps={
    Object {
      "className": "you shall pass",
    }
  }
>
  <section
    className="you shall pass show-on-desktop"
  >
    div
  </section>
</ShowOnDesktop>
`;
